internal/hcl/evaluator.go

```diff
@@ -111,22 +107,19 @@ func (e *Evaluator) Run() ([]*Module, error) {
 
 	// expand out resources and modules via count and evaluate again so that we can include
 	// any module outputs and or count references.
-	e.blocks = e.expandBlocks(e.blocks)
+	e.module.Blocks = e.expandBlocks(e.module.Blocks)
 	e.evaluate(lastContext)
 
-	// returns all the evaluated Blocks under their given Modules.
+	// returns all the evaluated Blocks under their given Module.
 	return e.collectModules(), nil
 }
 
-func (e *Evaluator) collectModules() []*Module {
-	rootModule := &Module{Blocks: e.blocks, RootPath: e.projectRootPath, ModulePath: e.modulePath}
-	modules := []*Module{rootModule}
-
+func (e *Evaluator) collectModules() *Module {
 	for _, definition := range e.moduleCalls {
-		modules = append(modules, definition.Modules...)
+		e.module.Modules = append(e.module.Modules, definition.Module)
```

Comment:
It seems currently `e.module.Modules` can be nil, if we don't always default it to an empty array, we probably should check it here. What do you think is the better approach?

Reply:
this should be fine, a nil slice is not the same as a nil struct see: https://go.dev/play/p/Rq1T8rdoV3j